File: src/bindings/gi/python3/testgi_kdb.py
import unittest
from gi.repository import GElektra as kdb

class Constants(unittest.TestCase):
	def setUp(self):
		pass

	def test_kdbconfig_h(self):
		self.assertIsInstance(kdb.DB_SYSTEM, str)
		self.assertIsInstance(kdb.DB_USER,   str)
		self.assertIsInstance(kdb.DB_HOME,   str)
		self.assertIsInstance(kdb.DEBUG,     int)
		self.assertIsInstance(kdb.VERBOSE,   int)

	def test_kdb_h(self):
		self.assertIsInstance(kdb.VERSION,       str)
		self.assertIsInstance(kdb.VERSION_MAJOR, int)
		self.assertIsInstance(kdb.VERSION_MINOR, int)
		self.assertIsInstance(kdb.VERSION_MICRO, int)
		self.assertIsNone(kdb.KS_END)

class KDB(unittest.TestCase):
	def setUp(self):
		pass

	'''
	disabled until elektra libraries can be loaded by
	the build system (and static build is gone)

	def test_ctor(self):
		self.assertIsInstance(kdb.KDB(), kdb.KDB)
		error = kdb.Key()
		self.assertIsInstance(kdb.KDB(error), kdb.KDB)

	def test_get(self):
		with kdb.KDB() as db:
			ks = kdb.KeySet()
			db.get(ks, "system/elektra")

			key = ks["system/elektra/version/constants/KDB_VERSION"]
			self.assertEqual(key.value, kdb.VERSION)

	def test_set(self):
		with kdb.KDB() as db:
			ks = kdb.KeySet(100)
			db.get(ks, "user/MyApp")

			try:
				key = ks["user/MyApp/mykey"]
			except KeyError:
				key = kdb.Key("user/MyApp/mykey")
				ks.append(key)
			key.value = "new_value"

			db.set(ks, "user/MyApp")

		with kdb.KDB() as db:
			ks = kdb.KeySet(100)
			db.get(ks, "user/MyApp")
			self.assertEqual(ks["user/MyApp/mykey"].value, "new_value")
	'''

if __name__ == '__main__':
	unittest.main()

File: src/bindings/gi/lua/testgi_kdb.lua
lgi = require('lgi')
kdb = lgi.require('GElektra')

TEST_NS = "user/tests/gi_lua"

-- kdbconfig.h
assert(type(kdb.DB_SYSTEM) == "string")
assert(type(kdb.DB_USER)   == "string")
assert(type(kdb.DB_HOME)   == "string")
assert(type(kdb.DEBUG)     == "number")

-- kdb.h
assert(type(kdb.VERSION) == "string")
assert(type(kdb.VERSION_MAJOR) == "number")
assert(type(kdb.VERSION_MINOR) == "number")
assert(type(kdb.VERSION_MICRO) == "number")
assert(kdb.KS_END == nil)

-- ctor
assert(kdb.KDB:is_type_of(kdb.KDB()))
local error = kdb.Key()
assert(kdb.KDB:is_type_of(kdb.KDB(error)))

-- get
do
	local db = kdb.KDB()
	local ks = kdb.KeySet(100)
	db:get(ks, "system/elektra")

	if os.getenv("CHECK_VERSION") == nil then
		local key = ks:lookup("system/elektra/version/constants/KDB_VERSION")
		assert(key.value == kdb.VERSION)
	end
end

-- set
do
	local db = kdb.KDB()
	local ks = kdb.KeySet(100)
	db:get(ks, TEST_NS)

	local key = ks:lookup(TEST_NS .. "/mykey")
	if not key then
		key = kdb.Key(TEST_NS .. "/mykey")
		ks:append(key)
	end
	key.string = "new_value"

	db:set(ks, TEST_NS)
end

do
	local db = kdb.KDB()
	local ks = kdb.KeySet(100)
	db:get(ks, TEST_NS)
	assert(ks:lookup(TEST_NS .. "/mykey").value == "new_value")
end

-- cleanup
do
	local db = kdb.KDB()
	local ks = kdb.KeySet(100)
	db:get(ks, TEST_NS)
	ks:cut(kdb.Key(TEST_NS))
	db:set(ks, TEST_NS)
end
